src/de/mod.rs

```diff
@@ -852,6 +852,12 @@ mod tests {
         assert_eq!(serde_json::from_str::<Empty>(r#"{}"#).unwrap(), Empty {});
     }
 
+    #[test]
+    fn unit() {
```

Comment:
Looks good.
Can you add another test for the variant issue as well (this may fix the reported error, but not tested here).

If you make

```rust
struct ContractResult {
  Ok(()),
  Err(String),
}
```

And test both the Ok and Err variants encode/decode properly, then I would be very confident in this (that all works in last release except `ContractResult::Ok` decode